lightning/src/routing/router.rs

```diff
@@ -1476,6 +1476,21 @@ where L::Target: Logger {
 	}
 
 	// Step (5).
+	// We only consider paths shorter than our maximum length estimate. 
+	// In the legacy onion format, the maximum number of hops used to be a fixed value of 20.
+	// However, in the TLV onion format, there is no fixed maximum length, but the `hop_payloads`
+	// field is always 1300 bytes. As the `tlv_payload` for each hop may vary in length, we have to
+	// estimate how many hops the route may have so that it actually fits the `hop_payloads` field.
+	//
+	// We estimate 2+8 (amt_to_forward) + 2+4 (outgoing_cltv_value) + 2+8 (short_channel_id) = 26
```

Comment:
Are these the right estimates? Am I missing something?

Reply:
Arg, no public key as far as I see, but I forgot to consider the `bigint: length` and the 32 byte `HMAC` that wrap each `hop_payload`...

Reply:
Yeah, nice to do all these calculations to arrive at 20 again.. :grin: (https://github.com/lightningdevkit/rust-lightning/pull/1476/commits/ef962dc7bb61668ebedf79c5fde554c5be2bdc12)

Reply:
> Cc @niftynei you sure it's 21?

I worded that question poorly; `21` is the max *nodes* in a route, there's 20 *hops* in an onion. Proof, from the spec

```
a 1300-byte hop_payloads consisting of multiple, variable length, hop_payload payloads or up to 20 fixed sized legacy hop_data payloads.
```